sklego/preprocessing.py

```diff
@@ -206,3 +206,179 @@ def _check_X_for_type(X):
         """Checks if input of the Selector is of the required dtype"""
         if not isinstance(X, pd.DataFrame):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
+
+
+class ColumnCapper(TransformerMixin, BaseEstimator):
+    """
+    Caps the values of columns according to the given quantile thresholds.
+
+    :type min_quantile: float, optional, default=0.05
+    :param min_quantile: The minimum quantile to compute the lowerbound of the transformed
+        columns. Must be in the interval [0; 1].
+
+    :type max_quantile: float, optional, default=0.95
+    :param max_quantile: The maximum quantile to compute the upperbound of the transformed
+        columns. Must be in the interval [0; 1].
+
+    :type discard_infs: bool, optional, default=False
+    :param discard_infs: Whether to discard ``-np.inf`` and ``np.inf`` values or not. If
+        ``False``, such values will be capped. If ``True``, they will be replaced by
+        ``np.nan``.
+
+        .. note::
+            Setting ``discard_infs=True`` is important if the `inf` values are results
+            of divisions by 0, which are interpreted by ``pandas`` as ``-np.inf`` or
+            ``np.inf`` depending on the signal of the numerator.
+
+    :raises:
+        ``TypeError`` if the quantiles are not numbers
+
+        ``ValueError`` if ``min_quantile`` > ``max_quantile`` or if the quantiles are
+        not in the interval [0; 1]
+
+    :Example:
+
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> from sklego.preprocessing import ColumnCapper
+    >>> df = pd.DataFrame({'a':[1, 2, 3, 4], 'b':[11, 12, np.inf, 14]})
+    >>> df
+       a     b
+    0  1  11.0
+    1  2  12.0
+    2  3   inf
+    3  4  14.0
+    >>> capper = ColumnCapper(min_quantile=.05, max_quantile=.9, discard_infs=False)
+    >>> df['a_capped'] = capper.fit_transform(df['a'])
+    >>> df['b_capped'] = capper.fit_transform(df['b'])
+    >>> df
+       a     b  a_capped  b_capped
+    0  1  11.0      1.15      11.1
+    1  2  12.0      2.00      12.0
+    2  3   inf      3.00      13.6
+    3  4  14.0      3.70      13.6
+    >>> capper = ColumnCapper(discard_infs=True) # Discarding infs
+    >>> df[['a', 'b']] = capper.fit_transform(df[['a', 'b']]) # Transforming multiple columns
+    >>> df
+          a     b  a_capped  b_capped
+    0  1.15  11.1      1.15      11.1
+    1  2.00  12.0      2.00      12.0
+    2  3.00   NaN      3.00      13.6
+    3  3.85  13.8      3.70      13.6
```

Comment:
i wonder ... why are we not seeing `13.8` at `b_capped` here?

Reply:
I also notice that `np.inf` values are filtered out during fit but `NaN` values are not. Is there a reason for this?

Reply:
Hello, @koaning 

### On the examples

> i wonder ... why are we not seeing `13.8` at `b_capped` here?

I used different quantiles for the second capping... maybe it was not an intuitive move. Do you think I should use the same capping quantiles or keep them different and point out to the fact that I changed the capping values?

### On *inf* cells

> I also notice that `np.inf` values are filtered out during fit but `NaN` values are not. Is there a reason for this?

Yes. The *inf* part came from [here](https://github.com/koaning/scikit-lego/issues/66#issuecomment-486374288) (typo fixed):

> This ColumnCapper could then take all `inf` values and cast them to the 95th percentile and `-inf` to the 5th.

And from [here](https://github.com/koaning/scikit-lego/issues/66#issue-424303629):

> In ML some time your features have extreme large values or even infinite value (np.inf), we want to cap those values with a feature transformer.

Leaving *inf* cells untouched doesn't make much sense to me since we're attempting to guarantee capping limits. But this is just my point of view...

As an attempt to contemplate what I said [here](https://github.com/koaning/scikit-lego/issues/66#issuecomment-486225626):

> I usually replace them by np.nan and then deal with those from there.

I made it an option with the `discard_infs` parameter and pointed out the importance of using it depending on how those *inf* cells were produced. But its default value is `False`, which makes `ColumnsCapper` simply cap them.

### On *nan* cells

Unfortunately, we can't know what to do with the *nan* cells, as you said [here](https://github.com/koaning/scikit-lego/issues/66#issuecomment-486374288):

> `Na` remains tricky, but thats what valueimputers are for.

So I left those out as a responsibility for the user. Enforcing some artificial value on them with this transformer doesn't make much sense to me. But again, this is just my point of view. What do you think?

Reply:
Hello @koaning!

> i wonder ... why are we not seeing `13.8` at `b_capped` here?

I used different quantiles for the second capping... maybe it was not an intuitive move. Do you think I should use the same capping quantiles or keep them different and point out to the fact that I changed the capping values?

> I also notice that np.inf values are filtered out during fit but NaN values are not. Is there a reason for this?

Yeah, `pandas` ignores *nan* cells when computing the quantiles. But if I leave *inf* cells there, they will pull the quantiles towards them.

```python
>>> import pandas as pd
>>> import numpy as np
>>> df = pd.DataFrame([1, 2, 3, 4, 5, np.inf])
>>> df.quantile([0.05, 0.95])
         0
0.05  1.25
0.95   inf
```

Reply:
That makes sense